102.md

```diff
@@ -0,0 +1,96 @@
+NIP-102
+=======
+
+Hierarchical Deterministic Key Management
+-----------------------------------------
+
+`draft` `optional`
+
+This NIP defines a way to separate identity from authentication using hierarchical deterministic (HD) keys. This allows people to use one key pair to issue independent key pairs for different apps. If a key is compromised, the root key pair can publish an event revoking that key as of a specific time.
+
+HD keys (BIP-32, BIP-39, BIP-44) provide a means of creating new key pairs from a parent in such a way that the new pubkey can be verified to be a child of the parent's pubkey. A message signed by this new key can be rapidly and locally verified as an authentic subkey of the claimed parent.
+
+NIP-06 declares the default nostr derivation path to be `m/44'/1237'/<account>'/0/0`.
+
+```
+recovery phrase -> seed -> xpriv
+   -> m/44'/1237'/0'     -> Account0
+   -> m/44'/1237'/0'/0/0 -> Account0 Subkey0
+   -> m/44'/1237'/0'/1/0 -> Account0 Subkey1
+   -> m/44'/1237'/0'/2/0 -> Account0 Subkey2
+```
+
+## Management Event
+
+The parent key can publish a `Kind 10102` event for subkey management. This event lists subkeys that have been revoked, as well as those that are currently active, and a preference for how valid subkeys that are not listed should be treated. This is only a preference, as it may not always be immediately available to clients and relays.
+
+Content:
+```json
+{
+  "keys": {
+    "<hex-subkey0-pubkey>": { "active_at": "<epoch-timestamp>" },
+    "<hex-subkey1-pubkey>": { "active_at": "<epoch-timestamp>", "revoked_at": "<epoch-timestamp>" },
+    "<hex-subkey2-pubkey>": { "active_at": "<epoch-timestamp>" }
+  },
+  "default_policy": "allow"
+}
+```
```

Comment:
I'd model this using single-letter tags so that they can be indexed. That would then obviate the need for adding an `account` tag to every event published by a subkey, since client could just fetch any `10102`s for pubkeys they don't recognize.

Reply:
I'm not sure that I understand how using a single letter tag would change whether or not we need a tag with the `account`s npub. My suspicion is that the pipeline stalls caused by constantly fetching new `10102`s would outweigh the space savings. Is the efficiency of single-letter vs word tags due to how relays are optimized?